### PR TITLE
creation:saving publish date in tutorial resource

### PR DESCRIPTION
**Pull request by kirti3192**

### CODE REVIEW

File: creation/views.py [hunk 1]
```diff
@@ -2114,9 +2115,11 @@ def publish_tutorial(request, trid):
     else:
         flag = 1
     if flag and tr_rec.outline_status == 4 and tr_rec.script_status == 4 and tr_rec.video_status == 4:
-        tr_rec.status = 1
+        tr_rec.status = 1 
+        tr_rec.publish_date = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
```
Comment (theskumar) [inline creation/views.py]: Let change this to:
```
from django.utils import timezone

tr_rec.publish_at = timezone.now()
```

File: creation/models.py [hunk 2]
```diff
@@ -196,6 +196,7 @@ class TutorialResource(models.Model):
     hit_count = models.PositiveIntegerField(default=0)
     created = models.DateTimeField(auto_now_add=True)
     updated = models.DateTimeField(auto_now=True)
+    publish_date = models.DateTimeField(null=True)
```
Comment (theskumar) [inline creation/models.py]: rename to `publish_at`

File: creation/models.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # Third Party Stuff
 from django.contrib.auth.models import User
 from django.db import models
-
+from django.utils import timezone
```
Comment (theskumar) [inline creation/models.py]: Remove this, not needed.

File: cron/add_publish_date.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
+from django.db.models import Q
```
Comment (theskumar) [inline cron/add_publish_date.py]: Move this to management command. e.g. https://github.com/Spoken-tutorial/spoken-website/blob/eventsV2/base/management/commands/initial_data.py

Or just have it run once on the production server inside django shell.

File: cron/add_publish_date.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+from creation.models import TutorialResource, PublishTutorialLog
+
+with open(LOG_ROOT+'publish_date_log.txt',"w") as success_log_file_head:
+	count =0 
+
+	tutorials = TutorialResource.objects.all()
+	for tutorial in tutorials:
+		count = count + 1
+		try:
+			latest_publish_date = PublishTutorialLog.objects.filter(tutorial_resource_id = tutorial.id).order_by('-id').first()
```
Comment (theskumar) [inline cron/add_publish_date.py]: log_obj = PublishTutorialLog.objects.filter(tutorial_resource=tutorial, created__isnull=False).order_by('-id').first()
if log_obj:
  tuturial.publish_at = log_obj.created

**Comment by kirti3192** [comment]
@theskumar : All requested changes are done. Plz have a look at it

**Pull request merged**